### PR TITLE
feat: add applications subdir

### PR DESCRIPTION
**Pull request by ldming**
* Add all addon CRs in KubeBlocks helm chart
* Add `applications` sub directory
* Rename `plugins` to `cli-plugins`

### CODE REVIEW

File: addons/apecloud-mysql/apecloud-mysql-addon.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    helm.sh/resource-policy: keep
+  {{- end }}
+spec:
+  description: 'ApeCloud MySQL is a database that is compatible with MySQL syntax and achieves high availability
+    through the utilization of the RAFT consensus protocol.'
+
+  type: Helm
+
+  helm:
+    {{- include "kubeblocks.addonChartLocationURL" ( dict "name" "apecloud-mysql" "version" (default .Chart.Version .Values.versionOverride) "values" .Values) | indent 4 }}
```
Comment (1aal) [inline addons/apecloud-mysql/apecloud-mysql-addon.yaml:19]: Do we need to make corresponding changes here when dependencies on 'KubeBlock' are removed from the addon? Alternatively, should we add a readme in the subfolder like `\addons` or `\tools` to help third-party users understand the meanings of these fields

Comment (ldming) [reply addons/apecloud-mysql/apecloud-mysql-addon.yaml:19]: Yes, we need. This PR only move addon CRs to this repo, more works will be done in the next PRs.

Comment (1aal) [reply addons/apecloud-mysql/apecloud-mysql-addon.yaml:19]: > Yes, we need. This PR only move addon CRs to this repo, more works will be done in the next PRs.

Get it!

**Review by 1aal — APPROVED** [review]

**Comment by fengluodb** [comment]
If rename `plugins` to `cli-plugins`, the kbcli plugin cmd needs adaptation. The plugin cmd will also need to search `cli-plugins` dir.    We need a issue to mark it when this pr is merged.

**Review by fengluodb — APPROVED** [review]

**Pull request merged**